docs/advanced/accesstransformers.md

```diff
@@ -0,0 +1,111 @@
+Access Transformers
+===================
+
+Access Transformers (ATs for short) allows for changing the visibility and `final` flags of classes and their methods and fields. They allow modders to access and modify otherwise `private` members in classes outside their control.
+
+The [specification document][specs] can be viewed on the Minecraft Forge GitHub.
+
+!!! note
+    An access transformer preprocessor is required to compile against access-transformed code. ForgeGradle fulfills this purpose for Forge development.
+
+Adding ATs
+----------
+
+Adding an Access Transformer to your mod project is as simple as adding a single line into your `build.gradle`:
+
+```groovy
+// This block is where your mappings version is also specified
+minecraft {
+    accessTransformer = file('src/main/resources/META-INF/accesstransformer.cfg')
+}
+```
+
+During development, the AT file can be anywhere specified by the line above. However, when loading in a non-development environment, Forge will only search for the exact path of `META-INF/accesstransformer.cfg` in your JAR file.
+
+Comments
+--------
+
+All text after a `#` until the end of the line will be treated as a comment and will not be parsed.
+
+Access Modifiers
+----------------
+
+Access modifiers specify which member visibility the given target will be transformed to. In decreasing order of visibility:
+
+  * `public` - visible to all classes inside and outside its package
+  * `protected` - visible only to classes inside the package and subclasses
+  * `default` - visible only to classes inside the package
+  * `private` - visible only to inside the class
+
+A special modifier `+f` and `-f` can be appended to the aforementioned modifiers to either add or remove respectively the `final` modifier, which prevents subclassing, method overriding, or field modification when applied.
+
+!!! warning
+    Access transformers can **_never_** make a member less visible; they can only widen visibility. The JVM will throw an `IllegalAccessError` if any compiled code references a method or field which cannot be accessed due to restrictive transformations.
+    
+    Directives only modify the method they directly reference; any overriding methods will _not_ be access-transformed. Therefore, **never** transform any non-`public` or non-`static` method, because any overriding methods might still use the restrictive visibility, which will result in the JVM throwing a `VerifyError`. 
```

Comment:
You can make private methods public to call them, because they cannot be overridden anywhere.

Reply:
Same applies to final methods, no matter the visibility.

Reply:
Ah, must've been confused when writing that piece. WIll correct it.

Reply:
> Same applies to final methods, no matter the visibility.

Yeah, I'll need to think through the logic again. Thanks to both of you for correcting this.